locksmith/src/websub/helpers.ts

```diff
@@ -70,10 +71,34 @@ export async function notifyHook(hook: Hook, body: unknown) {
   hookEvent.topic = hook.topic
   hookEvent.body = JSON.stringify(body)
 
-  // Save the pending state in database
-  await hookEvent.save()
-
   try {
+    // Save the pending state in database
+    await hookEvent.save()
+
+    // Get last 3 HookEvents created after last hook update.
+    const previousHookEventsForCurrentHook = await HookEvent.findAll({
+      where: {
+        hookId: hook.id,
+        updatedAt: {
+          [Op.gte]: hook.updatedAt,
+        },
+      },
+      order: [['createdAt', 'DESC']],
+      limit: 3,
+    })
+
+    const checkBadHealth = previousHookEventsForCurrentHook.every(
+      (event) => event.state !== 'success'
+    )
+
+    // If three attempts were not successful, unsubscribe the hook and return the pending event
+    if (checkBadHealth) {
+      logger.warn(`Unsubscribing hook (${hook.id}) due to being unresponsive.`)
```

Comment:
Ha no, we should not unsubscribe, just stop retrying. The expiration date will fix this.
It is important because if for some reason a subscriber is down for a few minutes at the time of a notification we don't want to assume they're down "for good". If we did this they would need to resubscribe to everything once they come back up and that would cause extra load.

Reply:
okay!